src/test/proxy/worker_proxy.py

```diff
@@ -0,0 +1,253 @@
+from typing import List
+from unittest import TestCase
+
+from ja.common.job import Job, JobStatus, JobPriority, JobSchedulingConstraints
+from ja.common.docker_context import DockerContext, DockerConstraints
+from ja.common.message.worker import WorkerResponse
+from ja.server.proxy.proxy import IWorkerProxy
+from test.abstract import skipIfAbstract
+
+
+class WorkerProxyDummy(IWorkerProxy):
+    def __init__(self, uid: str, jobs: List[Job]):
+        self._uid = uid
+        self._jobs: List[Job] = []
+        for job in jobs:
+            job_copy = Job.from_dict(job.to_dict())
+            job_copy.status = JobStatus.RUNNING
+            self._jobs.append(job_copy)
+
+    def _find_job(self, uid: str) -> Job:
+        for job in self._jobs:
+            if job.uid == uid:
+                return job
+        return None
+
+    @property
+    def uid(self) -> str:
+        return self._uid
+
+    def dispatch_job(self, job: Job) -> WorkerResponse:
+        for existing_job in self._jobs:
+            if job.uid == existing_job.uid:
+                return WorkerResponse.from_dict(WorkerResponse(
+                    result_string=IWorkerProxy.DISPATCH_JOB_DUPLICATE % (job.uid, self.uid), is_success=False
+                ).to_dict())
+        job_copy = Job.from_dict(job.to_dict())
+        job_copy.status = JobStatus.RUNNING
+        self._jobs.append(job_copy)
+        return WorkerResponse.from_dict(WorkerResponse(
+            result_string=IWorkerProxy.DISPATCH_JOB_SUCCESS % (job.uid, self.uid), is_success=True).to_dict())
+
+    def cancel_job(self, uid: str) -> WorkerResponse:
+        for job in self._jobs:
+            if uid == job.uid:
+                job.status = JobStatus.CANCELLED
+                self._jobs.remove(job)
+                return WorkerResponse.from_dict(WorkerResponse(
+                    result_string=IWorkerProxy.CANCEL_JOB_SUCCESS % (uid, self.uid), is_success=True).to_dict())
+        return WorkerResponse.from_dict(WorkerResponse(
+            result_string=IWorkerProxy.CANCEL_JOB_UNKNOWN_JOB % (uid, self.uid), is_success=False).to_dict())
+
+    def pause_job(self, uid: str) -> WorkerResponse:
+        for job in self._jobs:
+            if uid == job.uid:
+                try:
+                    job.status = JobStatus.PAUSED
+                    return WorkerResponse.from_dict(WorkerResponse(
+                        result_string=IWorkerProxy.PAUSE_JOB_SUCCESS % (uid, self.uid), is_success=True).to_dict())
+                except ValueError:
+                    return WorkerResponse.from_dict(WorkerResponse(
+                        result_string=IWorkerProxy.PAUSE_JOB_NOT_RUNNING % (uid, self.uid), is_success=False
+                    ).to_dict())
+        return WorkerResponse.from_dict(WorkerResponse(
+            result_string=IWorkerProxy.PAUSE_JOB_UNKNOWN_JOB % (uid, self.uid), is_success=False).to_dict())
+
+    def resume_job(self, uid: str) -> WorkerResponse:
+        for job in self._jobs:
```

Comment:
not sure how much functional-style python you want to use, this could be `next(job for job in self._jobs if job.uid == uid)`

Reply:
of course same for other functions, although I'm not sure what way is more pythonic

Reply:
I think what I currently have is easier to understand.